### DISCUSSION TITLE
Exceeded request rate limit. Please try again in a few hours. Alternatively, inject an instance of HttpClient that includes cookies for authenticated user.

### DISCUSSION THREAD

**Discussion by kirill2400** [Tyrrrz/YoutubeExplode#570] [General]
`
Exception: Exceeded request rate limit. Please try again in a few hours. Alternatively, inject an instance of HttpClient that includes cookies for authenticated user.
`

What can i do?

**Comment by Tyrrrz** [comment]
Please try again in a few hours. Alternatively, inject an instance of HttpClient that includes cookies for authenticated user.

Some similar issues: #453 #376 #292 #312 #273 #436 #321 #470 #500

**Reply by kirill2400** [reply]
It's been over a few days now...

**Reply by kirill2400** [reply]
Please tell me how to use httpclient with your api key? I have tried via google SDK and native httpclient, does not accept

**Reply by Tyrrrz** [reply]
You need to log in to YouTube in your browser, extract all the cookies, and then set them manually on an instance of `CookieContainer` inside `HttpClientHandler` passed to `HttpClient`. Maybe this comment will help you with identifying what cookies you need: https://github.com/Tyrrrz/YoutubeExplode/issues/376#issuecomment-627324714
Here's how to set them: https://github.com/Tyrrrz/YoutubeExplode/issues/557#issuecomment-859671776

Otherwise there's not much you can do, rate limiting is done by YouTube.

**Reply by kirill2400** [reply]
ty, it's working
